src/k8s-engine/wsl.ts

```diff
@@ -1435,12 +1489,17 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
     const result: Record<string, boolean | string> = {};
 
     const executable = await this.getWSLHelperPath();
+    const registeredDistros = (await this.registeredDistros()).filter(distro => !DISTRO_BLACKLIST.includes(distro));
 
-    for (const distro of await this.registeredDistros()) {
-      if (DISTRO_BLACKLIST.includes(distro)) {
-        continue;
-      }
+    if (!this.#enabledK3s) {
+      registeredDistros.forEach((distro: string) => {
+        result[distro] = true;
+      });
 
+      return result;
```

Comment:
That seems really odd. Among other things, this gives odd errors on _unchecking_ the integration.  Also, this means I can't run _without_ integration (e.g. if I want to run my own distro-specific `dockerd`).

Reply:
Storing state in the config now.